### PR TITLE
Add allow for out of bounds indexing

### PR DESCRIPTION
**Pull request by yancyribbens**
Out of bounds indexing is a workaround for const panic until MSRV +1.57

### CODE REVIEW

**Comment by yancyribbens** [comment]
This fixes the new clippy lint.

**Comment by apoelstra** [comment]
Could you rebase this on master?

It is currently based on the tip of #2304 (not its merge commit) which makes the history slightly hard to follow. Github is smart enough to detect this as "one commit" but my homebrew tree-scanning software isn't.

**Comment by yancyribbens** [comment]
> Could you rebase this on master?

Oops, sorry about that.  Rebased on master.

**Review by apoelstra — APPROVED** [review]
ACK 278229def5d96d9b976a0c5e0777a3fa876d5932 Happy New Year! Will merge based on CI one-ack carveout.
